File: exercises/week02/src/MotionSensor.java
/**
 * Created by samaj on 21/01/17.
 */
public class MotionSensor extends SecuritySensor {
    private String location;
    private String sensorType;

    public MotionSensor(String location) {
        this.location = location;
        this.sensorType = "Motion sensor";
    }

    @Override
    public boolean isTriggered() {
        return Math.random() < 0.05;
    }

    @Override
    public String getLocation() { return location; }

    @Override
    public String getSensorType() { return sensorType; }
}

File: exercises/week02/src/MotionSensor.scala
/**
  * Created by samaj on 22/01/17.
  */

import scala.util.Random

class MotionSensor(location: String) extends SecuritySensor {
  private val sensorType: String = "Motion"

  override def isTriggered: Boolean = Random.nextDouble() < 0.1

  override def getLocation: String = location

  override def getSensorType: String = sensorType

}
